scss/bootstrap/variants/_checkbox-switch.scss

```diff
@@ -0,0 +1,118 @@
+.checkbox-switch {
+  padding-top: 0 !important;
+  position: relative;
+
+  input {
+    bottom: 0;
+    height: 0;
+    left: 0;
+    margin: 0;
+    right: 0;
+    top: 0;
+    visibility: hidden;
+    width: 0;
+
+    &:checked {
```

Comment:
proposing a solution based on some more advanced usage of sass. Given that all that `&:checked` and `&[disabled]` do is to affect `.checkbox-switch-label`, what about leveraging `&` to move those styles under `.checkbox-switch-label`:
```scss
.checkbox-switch {
  // ...

  input {
    bottom: 0;
    // ...
    width: 0;
  }
}

.checkbox-switch-label {
  // ..
  input:checked + & {
    &::after {
      // ...
    }

    &::before {
      // ...
    }
  }

  input[disabled] + & {
    // ...
  }
}
```